BrowserKit/Sources/ToolbarKit/AddressToolbar/LocationView/LocationView.swift

```diff
@@ -119,6 +118,11 @@ final class LocationView: UIView, LocationTextFieldDelegate, ThemeApplicable, Ac
         self.isUnifiedSearchEnabled = isUnifiedSearchEnabled
         searchTerm = state.searchTerm
         onLongPress = state.onLongPress
+
+        searchEngineContentView = isUnifiedSearchEnabled
+                                  ? DropDownSearchEngineView()
+                                  : PlainSearchEngineView()
```

Comment:
I'm wondering if it's a good idea to check for `isUnifiedSearchEnabled` every time the state changes. Wouldn't it be better to check at the initialization of `searchEngineContentView`?

Reply:
Good catch, you're right, this does get fired a lot! Didn't realize it got triggered with every Toolbar state update. I will add some additional business logic so we just switch the views here, instead of rendering new ones. What do you think?

Unfortunately, I don't think we can leverage the init for this. We have to pass in the `isUnifiedSearchEnabled` Nimbus flag from a parent view (BrowserKit can't access the flags from Client directly). As a result, we don't know whether `isUnifiedSearchEnabled` is enabled or disabled yet when LocationView's init is called. 

I could make a new convenience init for LocationView which takes `isUnifiedSearchEnabled`, but then we still run the risk of someone calling `.build` (as we do with most of our UIView subclasses) and bypassing the convenience init.

Reply:
Yes, I believe the approach of having the views already initialized and just switching between them based on the flag is the best in this case.

Reply:
Yes, I believe that the approach of having the views already initialized and switching between them based on the flag, is the best in this case.